### PR TITLE
Disentangle cola

### LINKED ISSUE #432: Modify the CoLA dataset so that it can be used with a variety of models.
The CoLA example does not currently have a clear separation between model and dataset. Instantiating an instance of `CoLA` involves [providing a pre-trained BERT model](https://github.com/tensorflow/swift-models/blob/master/Examples/BERT-CoLA/main.swift#L38-L42). This could be clearer by separating the preprocessing functionality, similar to the way [`TextUnsupervised` requires a `BytePairEncoder`](https://github.com/tensorflow/swift-models/blob/master/Examples/GPT2-WikiText2/main.swift#L27-L29), but not an entire pre-trained GPT2 model.

The CoLA dataset looks close to being model-independent, but this should be reflected in the example. Moving some of the functionality from CoLATraining.swift into more reusable places would align it well with other existing examples and datasets.

Longer-term, it would likely make sense to introduce more general structures for preprocessing functionality, as our collection of transformer variants increases.

### PR DESCRIPTION
**Pull request by texasmichelle**
Partially addresses #432 

Moves training and eval functions from extensions to CoLA to the main example file.
Adds more output to training and eval loops.
Removes print statements from MCC function.

### CODE REVIEW

**Comment by texasmichelle** [comment]
I'd like to find a better way of separating BERT from CoLA. Exposing `exampleMapFn` in the main example is overly complex. Suggestions for where to tuck this away?

I'd also like to understand how this batching technique could fit in with Batcher. The bucketing approach would be useful to have in a more generic place. As is, the terminology is confusing since they operate in very different ways (see [Batchable](https://github.com/tensorflow/swift-models/blob/master/Datasets/CoLA/Batchable.swift) and the [hierarchy of iterators](https://github.com/tensorflow/swift-models/blob/master/Datasets/CoLA/DataUtilities.swift)).

File: Examples/BERT-CoLA/main.swift [hunk 1]
```diff
@@ -54,11 +72,64 @@ var optimizer = WeightDecayedAdam(
...
-    if step > 0 && step.isMultiple(of: 10) {
-        print("Evaluate BERT for the CoLA task:")
-        print(colaTask.evaluate(using: bertClassifier))
+for epoch in 1...10 {
+    print("[Epoch \(epoch)]")
+    Context.local.learningPhase = .training
+    var trainingLossSum: Float = 0
+    var trainingBatchCount = 0
+
+    for _ in 1...10 {
```
[the remark this answers is not in the document]
Comment (texasmichelle) [reply Examples/BERT-CoLA/main.swift:81]: Ah, that explains why my attempts to loop over this dataset were failing. I'll update 👍

File: Examples/BERT-CoLA/main.swift [hunk 2]
```diff
@@ -54,11 +72,64 @@ var optimizer = WeightDecayedAdam(
...
+        )
+    }
+
+    Context.local.learningPhase = .inference
+    var testLossSum: Float = 0
+    var testBatchCount = 0
+    var devDataIterator = cola.devDataIterator
+    var devPredictedLabels = [Bool]()
+    var devGroundTruth = [Bool]()
+    while let batch = withDevice(.cpu, perform: { devDataIterator.next() }) {
```
[the remark this answers is not in the document]
Comment (texasmichelle) [reply Examples/BERT-CoLA/main.swift:109]: I exposed the test iterator so that it's no longer private, but left this as dev since the test iterator hides labels and we need them to calculate MCC. Would you prefer to leave this as-is or replace with one of these options:
1. Use `testDataIterator` without MCC
1. Modify `testDataIterator` to include labels and use it here with MCC

[the remark this answers is not in the document]
Comment (xihui-wu) [reply Examples/BERT-CoLA/main.swift:109]: I see!  A second look, no problem using dev here. (my bad I thought this is outside epoch loop).

**Comment by texasmichelle** [comment]
@sgugger these changes should make it a bit easier to add Batcher
